File: frontend/react/src/types/companyIntelligence.ts
// Mirrors backend/schemas/company_intelligence.py's CompanyIntelligenceResponse
// (V3 Phase 7A). The nested `company` shape intentionally omits
// created_at/updated_at - the backend's CompanyOut schema doesn't
// include them either, unlike the full Company type in ./company.ts.

export interface IntelligenceCompany {
  id: string;
  name: string;
  industry: string | null;
  headquarters: string | null;
  website: string | null;
  monitoring_status: string;
}

export interface Technology {
  id: string;
  name: string;
  category: string | null;
  adoption_status: string | null;
  business_relevance: string | null;
  confidence_score: number | null;
  source: string | null;
}

export interface BusinessInitiative {
  id: string;
  name: string;
  description: string | null;
  category: string | null;
  priority: number | null;
  status: string | null;
  confidence_score: number | null;
}

export interface Executive {
  id: string;
  name: string;
  title: string | null;
  department: string | null;
  biography: string | null;
  linkedin_url: string | null;
  confidence_score: number | null;
}

export interface Signal {
  id: string;
  type: string;
  title: string;
  description: string | null;
  source: string | null;
  confidence: number | null;
  date_detected: string;
}

export interface GleanKnowledgeItem {
  source: string;
  content: string;
  category: string | null;
}

export interface CompanyIntelligence {
  company: IntelligenceCompany;
  technologies: Technology[];
  business_initiatives: BusinessInitiative[];
  executives: Executive[];
  recent_signals: Signal[];
  glean_knowledge: GleanKnowledgeItem[];
}

File: backend/schemas/company_intelligence.py
"""Response schema for GET /api/v1/companies/{company_id}/intelligence
(V3 Phase 7A). Serializes
backend/services/company_intelligence_service.py's
CompanyIntelligenceProfile (built entirely in Phase 5) - this endpoint
adds no new aggregation logic, only a read-only view over it.
"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class CompanyOut(BaseModel):
    id: str
    name: str
    industry: Optional[str] = None
    headquarters: Optional[str] = None
    website: Optional[str] = None
    monitoring_status: str

    model_config = {"from_attributes": True}


class TechnologyOut(BaseModel):
    id: str
    name: str
    category: Optional[str] = None
    adoption_status: Optional[str] = None
    business_relevance: Optional[str] = None
    confidence_score: Optional[float] = None
    source: Optional[str] = None

    model_config = {"from_attributes": True}


class BusinessInitiativeOut(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    category: Optional[str] = None
    priority: Optional[int] = None
    status: Optional[str] = None
    confidence_score: Optional[float] = None

    model_config = {"from_attributes": True}


class ExecutiveOut(BaseModel):
    id: str
    name: str
    title: Optional[str] = None
    department: Optional[str] = None
    biography: Optional[str] = None
    linkedin_url: Optional[str] = None
    confidence_score: Optional[float] = None

    model_config = {"from_attributes": True}


class SignalOut(BaseModel):
    id: str
    type: str
    title: str
    description: Optional[str] = None
    source: Optional[str] = None
    confidence: Optional[float] = None
    date_detected: datetime

    model_config = {"from_attributes": True}


class GleanKnowledgeItemOut(BaseModel):
    source: str
    content: str
    category: Optional[str] = None

    model_config = {"from_attributes": True}


class CompanyIntelligenceResponse(BaseModel):
    company: CompanyOut
    technologies: List[TechnologyOut]
    business_initiatives: List[BusinessInitiativeOut]
    executives: List[ExecutiveOut]
    recent_signals: List[SignalOut]
    glean_knowledge: List[GleanKnowledgeItemOut]
